frontend/src/components/signup.tsx

```diff
@@ -12,13 +12,23 @@ export const SignUp = () => {
     const [lastName, setLastName] = useState<string>('');
     const [email, setEmail] = useState<string>('');
     const [password, setPassword] = useState<string>('');
+    const [validLength, setValidLength] = useState<boolean>(false); 
+    const [hasNumber, setHasNumber] = useState<boolean>(false); 
+    const [specialChar, setSpecialChar] = useState<boolean>(false); 
     const [confirmPassword, setConfirmPassword] = useState<string>('');
     const [warning, setWarning] = useState<string>(''); 
+    const regex = /[\!\@\#\$\%\^\&\*\)\(\+\=\.\<\>\{\}\[\]\:\;\'\"\|\~\`\_\-]/g;
```

Comment:
let's use a more descriptive name for this var! reading this alone would tell me basically nothing about what regex pattern this is!